src/aihwkit/nn/modules/linear_sliced.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+
+# (C) Copyright 2020, 2021, 2022, 2023, 2024 IBM. All Rights Reserved.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Analog layers."""
+from typing import Any, Optional, Type, List
+
+from torch import Tensor
+from torch.nn import Linear, ParameterList
+
+from aihwkit.exceptions import ModuleError
+from aihwkit.nn.modules.base import AnalogLayerBase
+from aihwkit.simulator.parameters.base import RPUConfigBase
+from aihwkit.nn import AnalogLinear
+
+
+class AnalogLinearBitSlicing(AnalogLayerBase, Linear):
```

Comment:
Actually it would be better if you would just write a new module which inherits from `AnalogContainerBase` or simply from `Module` instead of `AnalogLayerBase`. This is because if `AnalogLinear` layers are used as children of an `AnalogLayer` which will cause issues. See for instance the RNN module that internally also use `AnalogLinear` layers,  see [here](https://github.com/IBM/aihwkit/blob/69e524fbc2c5462cbe1a12ee1662879c001c30c0/src/aihwkit/nn/modules/rnn/layers.py#L20) and  [here](https://github.com/IBM/aihwkit/blob/69e524fbc2c5462cbe1a12ee1662879c001c30c0/src/aihwkit/nn/modules/rnn/rnn.py#L128).

Reply:
It might also not necessary to inherit from `Linear` at all, as this will cause additional issues with the `Linear` weight etc.